### PR TITLE
WIP: Some Zoe API suggestion as PR against a PR:

### PR DESCRIPTION
**Pull request by erights**

### CODE REVIEW

File: packages/dapp-simple-exchange/contract/deploy-autoswap.js [hunk 1]
```diff
@@ -132,15 +132,13 @@ export default async function deployContract(
   async function addExchangeOffer(offerAmount, payments, wantAmount) {
     const offerRules = makeRulesOfferXForY(offerAmount, wantAmount);
     const { invite: anInvite } = await publicAPI~.makeInvite();
-    const { seat } = await homeP~.zoe~.redeem(anInvite, offerRules, payments);
-    return seat~.addOrder();
+    return homeP~.zoe~.redeem(anInvite, offerRules, payments)~.offerResult;
```
Comment (katelynsills) [inline packages/dapp-simple-exchange/contract/deploy-autoswap.js:135]: I think this is outside the scope of this PR

File: packages/zoe/src/contracts/helpers/zoeHelpers.js [hunk 2]
```diff
@@ -5,6 +5,9 @@ import { sameStructure } from '@agoric/same-structure';
 export const defaultRejectMsg = `The offer was invalid. Please check your refund.`;
 export const defaultAcceptanceMsg = `The offer has been accepted. Once the contract has been completed, please check your payout`;
 
+// Must not escape
+const OfferResultPumpkin = harden({});
```
Comment (katelynsills) [inline packages/zoe/src/contracts/helpers/zoeHelpers.js:9]: I really like the simplifications in `makeInvitePair`, but I'm not sure I like the `OfferResultPumpkin` as shared data. Could we put it inside the `makeInvitePair` function?

File: packages/zoe/src/contracts/helpers/zoeHelpers.js [hunk 3]
```diff
@@ -41,6 +44,34 @@ export const makeZoeHelpers = zoe => {
     return sameStructure(getKeys(actual), expectedKeys);
   };
   const helpers = harden({
+    makeInvitePair(seatFn, customProperties = undefined, expected = undefined) {
+      let offerResult = OfferResultPumpkin;
+      let offerError = OfferResultPumpkin;
+      const seat = harden({
+        makeOffer: () => {
```
Comment (katelynsills) [inline packages/zoe/src/contracts/helpers/zoeHelpers.js:51]: What if the seat method isn't makeOffer? Should we parameterize this?

File: packages/zoe/test/swingsetTests/zoe/vat-bob.js [hunk 4]
```diff
@@ -133,7 +133,7 @@ const build = async (E, log, zoe, issuers, payments, installations, timer) => {
         bobPayments,
       );
 
-      const bobOutcome = await E(seat).exercise();
+      const bobOutcome = await E(seat).makeOffer();
```
Comment (katelynsills) [inline packages/zoe/test/swingsetTests/zoe/vat-bob.js]: Hmm, I don't like changing all of these to `makeOffer`. That seems to defeat the purpose of having a seat.

**Review by katelynsills — CHANGES_REQUESTED** [review]
I like the simplification, but I have some issues with the approach. Let's make sure to talk this over.

File: packages/dapp-simple-exchange/contract/deploy-autoswap.js (hunk 1 above)
Comment (erights) [reply packages/dapp-simple-exchange/contract/deploy-autoswap.js:135]: Look again. The old code redeems and calls `addOrder`. The new code redeems and uses `offerResult`.
